Assignment 1/src/main/java/com/cezaram28/Assignment1/repository/jdbc/JdbcAnswerRepository.java

```diff
@@ -0,0 +1,105 @@
+package com.cezaram28.Assignment1.repository.jdbc;
+
+import com.cezaram28.Assignment1.entity.Answer;
+import com.cezaram28.Assignment1.entity.Question;
+import com.cezaram28.Assignment1.entity.User;
+import com.cezaram28.Assignment1.repository.AnswerRepository;
+import lombok.RequiredArgsConstructor;
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.jdbc.core.simple.SimpleJdbcInsert;
+
+import javax.swing.text.html.Option;
+import java.util.*;
+
+@RequiredArgsConstructor
+public class JdbcAnswerRepository implements AnswerRepository {
+
+    private final JdbcTemplate template;
+
+    @Override
+    public List<Answer> findAll() {
+        List<Answer> answers = template.query("SELECT * FROM answer", new AnswerMapper());
+        for(Answer a : answers) {
+            JdbcUserRepository jdbcUserRepository = new JdbcUserRepository(template);
+            Optional<User> u = jdbcUserRepository.findById(a.getAuthor().getId());
+            JdbcQuestionRepository jdbcQuestionRepository = new JdbcQuestionRepository(template);
+            Optional<Question> q = jdbcQuestionRepository.findById(a.getQuestion().getId());
+            if(u.isPresent())
+                a.setAuthor(u.get());
+            if(q.isPresent())
+                a.setQuestion(q.get());;
+        }
+        return answers;
+    }
+
+    @Override
+    public Optional<List<Answer>> findUsersAll(int userId) {
```

Comment:
[Optional] Returning an Optional of List is usually not needed. Typically, you would return simply an empty list instead of an empty optional (and you would never return null instead of a list).